File: Api parcial/Api parcial/src/Models/Producto.js
import { DataTypes } from 'sequelize';
import db from '../Database/db.js';

const Producto = db.define('Producto', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  nombre: {
    type: DataTypes.STRING,
    allowNull: false
  },
  descripcion: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  vendedor: {
    type: DataTypes.STRING,
    allowNull: false 
  },
}, {
  timestamps: false
});

export default Producto;

File: lib/producto.dart
class Producto {
  final int id;
  final String nombre;
  final String vendedor;
  final int calificacion;

  Producto(this.id, this.nombre, this.vendedor, this.calificacion);
}

class FavItem {
  final int id;
  final String nombre;
  final String vendedor;
  final int calificacion;

  FavItem(this.id, this.nombre, this.vendedor, this.calificacion);

  Map<String , dynamic> toMap() {
    return {
      'id': id,
      'nombre' : nombre,
      'vendedor' : vendedor,
      'calificacion' : calificacion,
    };
  }
}
